libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx

```diff
@@ -0,0 +1,299 @@
+      markedForDeletion?: boolean;
+    }[]
+  >([]);
+
+  const [deleteService] = useDeleteServiceRequestMutation();
+  const [createService] = useCreateNoteServiceRequestMutation();
+
+  const reset = async () => {
+    try {
+      const deleteServices = services.map((service) => ({
+        ...service,
+        markedForDeletion: true,
+      }));
+      const deleteServiceOthers = serviceOthers.map((service) => ({
+        ...service,
+        markedForDeletion: true,
+      }));
+
+      setServices(deleteServices);
+      setServiceOthers(deleteServiceOthers);
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  const submitServices = async () => {
+    const toCreateOtherServices = serviceOthers.filter(
+      (service) =>
+        service.title !== null && !service.id && !service.markedForDeletion
+    );
+    const toCreateServices = services.filter(
+      (service) =>
+        service.enum !== null && !service.id && !service.markedForDeletion
+    );
+
+    const toRemoveOtherServices = serviceOthers.filter(
+      (service) => service.markedForDeletion && !!service.id
+    );
+
+    const toRemoveServices = services.filter(
+      (service) => service.markedForDeletion && !!service.id
+    );
+
+    const toRemoveServicesWithOther = [
+      ...toRemoveOtherServices,
+      ...toRemoveServices,
+    ];
+
+    try {
+      for (const service of toRemoveServicesWithOther) {
+        if (service.id) {
+          try {
+            await deleteService({
+              variables: {
+                data: {
+                  id: service.id,
+                },
+              },
+            });
+          } catch (error) {
+            console.error(
+              `Error deleting service with id ${service.id}:`,
+              error
+            );
+          }
+        }
+      }
+
+      for (const service of toCreateServices) {
+        if (service.enum) {
+          try {
+            await createService({
+              variables: {
+                data: {
+                  service: service.enum,
+                  noteId,
+                  serviceRequestType: type,
+                },
+              },
+            });
+          } catch (error) {
+            console.error(
+              `Error creating service with enum ${service.enum}:`,
+              error
+            );
+          }
+        }
+      }
+
+      for (const service of toCreateOtherServices) {
+        if (service.title) {
+          try {
+            await createService({
+              variables: {
+                data: {
+                  service: ServiceEnum.Other,
+                  serviceOther: service.title,
+                  noteId,
+                  serviceRequestType: type,
+                },
+              },
+            });
+          } catch (error) {
+            console.error(
+              `Error creating service with title ${service.title}:`,
+              error
+            );
+          }
+        }
+      }
```

Comment:
oh many, now I know what you meant earlier. 
Could potentially be 25+ separate requests 🤣

Reply:
> potentially be 25+ separate requests
😳